src/test/ui/type/type-unsatisfiable.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
```

Comment:
I think this is OK with just build-pass, also can you rustfmt this file?

Reply:
If you have extra time, I would also be interested in adding a revision that tries to _use_ the code, like trying to subtract two `dyn Vector2` types.